src/core_plugins/kibana/public/context/api/utils/fields.js

```diff
@@ -0,0 +1,21 @@
+import _ from 'lodash';
+
+
+const addComputedFields = _.curry(function addComputedFields(indexPattern, query) {
```

Comment:
The motivation was to achieve the following goals:

* avoid duplication (DRY)
* assign a descriptive name to that reusable block of code
* enable independent testability

These are some of the very basic concepts of writing clean code. I think that function accomplishes all three. Whether to pass `indexPattern` or the `computedFields` is a matter of leaky abstractions. The function would still be coupled to the `IndexPattern` interface by virtue of assuming a particular data structure for the `computedFields`, so I preferred to avoid introducing another layer.